### DISCUSSION TITLE
Integrare Omhero in Home Assistant?

### DISCUSSION THREAD

**Discussion by mfabiani53** [indomus/forum#3043] [Nuove idee, proposte, progetti e guide]
Ho appena acquistato Omhero per controllare lo stato ed i costi dell'energia elettrica nella mia casa.
E' un progetto davvero interessante per cui vi chiedo se qualcuno ha provato o sta provando di integrare questo progetto dentro Home Assistant.
[edit]

**Comment by Squall87pd** [comment]
Piccolo up, anche io sarei interessato.

**Comment by Delpre89** [comment]
Mi accodo anch'io, vista l'accuratezza del dispositivo verrebbe molto comodo creare automazioni utilizzando i dati di OmHero (ex. Prycam Home), hanno già anche un cloud online da dove è possibile monitorare tutti i device, sarebbe sufficiente creare un integrazione con quell'account

**Comment by indomus** [comment]
Beh, verificate con loro se sia disponibile un set API cloud (o locale) per scrivere qualche componente di integrazione esterno, oppure se sia in roadmap la messa a disposizione dello standard Matter sui device (così da integrarlo usando quello).

**Comment by Squall87pd** [comment]
Allora, ho visto nelle impostazioni dell app di ohmero che è disponibile la trasmissione di dati tramite mqtt. Avendo home assistant dovrebbe essere facile integrare il tutto, ovviamente io di mqtt non mastico niente quindi devo mettermi li a cercare un po di guide e provare a smanettare un po, però con questo sistema si può fare l integrazione.

**Reply by Squall87pd** [reply]
Allora.. ho bisogno di qualcuno che mi segua passo passo per provare a configurare correttamente il broker mqtt e l'app Omhero.
Utilizzo Home Assistant e  ho seguito [questa guida](https://indomus.it/guide/configurare-mqtt-broker-client-su-home-assistant/). 

1. Creato l'utenza dedicata con user e pass.
2. installato il componente aggiuntivo Mosquitto Broker
3. configurato usando gli stessi user e pass creati in precedenza
4. impostato un port trigger sulla porta 1883 sul mio router di casa

Nella foto allegata si vedono i parametri di impostazione dell'app Omhero
<a href="https://ibb.co/JRSS3pXq"><img src="https://i.ibb.co/JRSS3pXq/IMG-1692.png" alt="IMG-1692" border="0"></a>

in user e pass immagino che io debba utilizzare quelli creati in precedenza, in mqtt:// non ho capito che link devo mettere (nella guida c'è scritto "l'indirizzo di interfaccia di loopback - 127.0.0.1" ma non so se sia corretto).

Andando invece ad impostare su home assitant un nuovo dispositivio mqtt, in impostazione mqtt non ho toccato nulla, mentre nella scheda "ascolta un argomento" ho impostato il parametro riportato nell'app.
<a href="https://ibb.co/d0N8N24F"><img src="https://i.ibb.co/d0N8N24F/Screenshot-2025-04-22-185117.png" alt="Screenshot-2025-04-22-185117" border="0"></a>

In ogni caso cosi facendo, non ho nessun dispositivo associato, premendo sui tre pallini - addmqtt devices, mi chede un configuration URL e anche qua non so che indirizzo andare a mettere (127.0.0.1 non è valido).
Immagino che da qualche parte io debb aandare a inserire sia l'ip del mio home assistant che del mio dispositivo Omhero che ho a disposizione trovati tramite l'app Fing sul pc .
<a href="https://ibb.co/YwTtDPy"><img src="https://i.ibb.co/YwTtDPy/Screenshot-2025-04-22-185504.png" alt="Screenshot-2025-04-22-185504" border="0"></a>

Inoltre, poi non saprei come andare avanti a configurare il tutto, mi chiede 

- il tipo di entita (Notification/sensor/switch) (immagino io debba selezionare sensore? )
- la classe del device (energy?)
- lo stato di classe del sensore (measurement?)

<a href="https://ibb.co/5Zmyhpt"><img src="https://i.ibb.co/5Zmyhpt/Screenshot-2025-04-22-190203.png" alt="Screenshot-2025-04-22-190203" border="0"></a>

e in ultimo mi chiede a che topic associare il dispositivo, quello me lo fornisce l'app ma poi in "value template" non saprei minimamente cosa inserire. 
<a href="https://ibb.co/RkjG8Qz7"><img src="https://i.ibb.co/RkjG8Qz7/Screenshot-2025-04-22-190413.png" alt="Screenshot-2025-04-22-190413" border="0"></a>

Grazie a chi avrà la pazienza di starmi un po dietro. Credo che oltre a non aver ben capito i vari paramenti da impostare, mi manchi anche tutto il disocrso del template da configurare..

**Comment by Squall87pd** [comment]
Aggiornamento: sono riuscito a configurare mosquitto ed ora riesco a leggere il topic fornito da Omhero. Ora però dovrei creare un template per tradurre le informazioni e settare un sensore. Ma non avrei idea di come fare.

**Reply by sergiocard** [reply]
ciao, sono interessato. Ho contattato il supporto info@XXXXX e mi hanno detto di:
inserire 
mqtt: !include mqtt.yaml
dentro il file configuration.yaml di home assistant e mi hanno fornito il file mqtt.yaml
A questo punto cosa bisogna fare? Ho gia il broker mqtt installato per via di zigbee2mqtt.
Mi manca però capire come va configurata l'app omhero e come "vedere" il dispositivo su home assistant

**Reply by lexius** [reply]
puoi condividere file yaml che ti hanno fornito?

**Reply by sergiocard** [reply]
> puoi condividere file yaml che ti hanno fornito?

Ciao, questo è il file che mi hanno fornito. Tieni presente che il file va rinominato in mqtt.yaml
Tieni inoltre presente che dentro questo file, la stringa "PH01-8-1863" la dovresti sovrascrivere dappertutto con il tuo client id che trovi dentro l'app:
![image](https://github.com/user-attachments/assets/b6b0711b-476a-4a0a-9093-8e9f6ec39fd8)

[mqtt.yaml.txt](https://github.com/user-attachments/files/20427393/mqtt.yaml.txt)

**Reply by holocantus** [reply]
> Aggiornamento: sono riuscito a configurare mosquitto ed ora riesco a leggere il topic fornito da Omhero. Ora però dovrei creare un modello per tradurre le informazioni e impostare un sensore. Ma non avrei idea di come fare.

Come ci sei riuscito ?io sono  al tuo stesso punto prima che scrivessi che vi sei riuscito. mi potresti aiutare? Grazie infinite

**Comment by lexius** [comment]
Ciao ragazzi. Dopo potete fare cosi:
![IMG_0431](https://github.com/user-attachments/assets/a36d2bf4-c033-42db-a7d6-0fa948bfc0fc)
![IMG_0433](https://github.com/user-attachments/assets/32e2a106-403f-4d7f-a455-442b945b8408)

**Reply by sergiocard** [reply]
ok ma tu scusa cosa hai messo dentro questi campi mqtt:// nome utente e password ? Qui dentro?
![image](https://github.com/user-attachments/assets/b52059f7-ce7f-4fd0-b690-1ab85035fe87)

**Reply by lexius** [reply]
![IMG_0435](https://github.com/user-attachments/assets/51eca18d-b82a-4f93-91d1-cdc0963cc874)
![IMG_0434](https://github.com/user-attachments/assets/f169fa7d-9ad0-475c-b9e0-3d423474b369)
Prima devi creare un utente in Home Assistant con accesso solo locale, come mqtt e password mqtt. Qui, visto che accesso sarà solo locale non serve inventare password complicato. Poi in impostazioni di server MQTT devi usare questo utente e password. Stessi dati usa in impostazioni di OMHERO per collegare al server mqtt. Indirizzo sarà IP del Home Assistant.

**Reply by lexius** [reply]
Ho guardato tuo yaml file e ho sistemato acumi i miei errori. Grazie
![IMG_0436](https://github.com/user-attachments/assets/a11f7a9e-0cff-4b2d-bfd8-9ebd722c2c62)

Adesso sembra che funzioni tutto, ma bisogna ancora fare un po’ di test riguarda del costo di energia

**Reply by it9tst** [reply]
Ciao @lexius, ottimo lavoro! Potresti condividere gentilmente il tuo file yaml con le migliorie applicate? Grazie

**Reply by lexius** [reply]
> Ciao @lexius, ottimo lavoro! Potresti condividere gentilmente il tuo file yaml con le migliorie applicate? Grazie

```
mqtt:
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    name: Ground Resistance
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 1.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_0cf672ff2ff141d0b1b5c6d2dd9c6456
    unit_of_measurement: "\u03A9"
    value_template: '{{ value_json.re }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: voltage
    name: Voltage
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 1.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_33257c0d610a4e3cb1714e499b32a24e
    unit_of_measurement: V
    value_template: '{{ value_json.vln / 1000 }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: reactive_power
    name: Reactive Power
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 1.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_36c91e42405046a98d457ee3ef4a00b6
    unit_of_measurement: var
    value_template: '{{ value_json.q / 1000 }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: temperature
    name: Temperature
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_397edfe770054afab190baf5542c3d26
    unit_of_measurement: "\xB0C"
    value_template: '{{ value_json.t }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: energy
    expire_after: 60
    name: Energia
    qos: 1.0
    state_topic: omhero/PH01-9-2513/meas
    suggested_display_precision: 3
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_6c64fde15cdf4e62a88a24e6c91b3b78
    unit_of_measurement: mWh
    value_template: '{{ value_json.e }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: power_factor
    name: Power Factor
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 2.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_78c788fcc58b4400bf60d7879f752366
    unit_of_measurement: '%'
    value_template: '{{ value_json.phi }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: current
    name: Current
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 1.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_9748da488e3141798ef9955650cc2c09
    unit_of_measurement: A
    value_template: '{{ value_json.il / 1000 }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    name: Alarm Over 130
    qos: 1.0
    state_topic: omhero/PH01-9-2513/tele
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_a1ae0524c63447cc951512ab0ee43062
    value_template: '{% if value_json.events[4] > 0 %} 0 {% else %} 1 {% endif %}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: frequency
    name: Frequency
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 2.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_b349e88d47be4c07a0dd08957042173c
    unit_of_measurement: Hz
    value_template: '{{ value_json.f }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: current
    name: Current diff.
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 1.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_b8756c09db404130bb2f2e1306f53b51
    unit_of_measurement: mA
    value_template: '{{ value_json.id }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: voltage
    name: Voltage Neutro/Ground
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 1.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_beed46efe09a4f9583fae4e5d0532a3f
    unit_of_measurement: V
    value_template: '{{ (value_json.vlg - value_json.vln) / 1000 }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    name: Alarm Over 100
    qos: 1.0
    state_topic: omhero/PH01-9-2513/tele
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_bf508214707d4229b1762c68d7c2938b
    value_template: '{% if value_json.events[3] > 0 %} 0 {% else %} 1 {% endif %}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: current
    expire_after: 60
    name: Omhero Current max
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/meas
    suggested_display_precision: 3
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_e4ddce5e6bdc4fdbbfcbc25927e89e83
    unit_of_measurement: A
    value_template: '{{ value_json.i_max }}'
- sensor:
    device:
      hw_version: '1.0'
      identifiers:
      - 01JVZF0N963NGK5FZB1SJK05MN
      model: PH01
      model_id: PH0100009000000000002513
      name: Omhero
      sw_version: 1.18-1.3.5-s
    device_class: power
    name: Active Power
    qos: 1.0
    state_class: measurement
    state_topic: omhero/PH01-9-2513/tele
    suggested_display_precision: 1.0
    unique_id: 01JVZF0N963NGK5FZB1SJK05MN_eb17e27cba384bccaf2374290d51c297
    unit_of_measurement: W
    value_template: '{{ (value_json.p / 1000) }}'

**Comment by lexius** [comment]
Ciao a tutti. Ho scoperto che OMHERO pubblica dati circa 2 volte al secondo in topic "tele" e una volta al minuto in topic "meas". Cioè telemetria e misurazione. Per adesso sto usando solo telemetria, ma in questo caso devo calcolare energia consumata da solo usando helper di home assistant - Integral Sensor. Non è molto comodo. Sembra che in topic "meas" c'è valore di energia, ma ho fatica di capire per quale periodo. Qualcuno mi può aiutare?
![IMG_0450](https://github.com/user-attachments/assets/aea6894b-4e22-4861-b60a-f381fbdc6bd1)
